api/config/dex/config_request.go

```diff
@@ -41,7 +41,11 @@ func DefaultConfigRequest() *ConfigRequest {
 	c.V1.Sys.Service.Host = w.String("127.0.0.1")
 	c.V1.Sys.Service.Port = w.Int32(10117)
 
-	c.V1.Sys.Expiry.IdTokens = w.String("3m")
+	if c.V1.Sys.Connectors.Saml != nil {
```

Comment:
:x: this doesn't seem to work as intended. Ideas?

Reply:
this will always be nil. this function is a constructor

Reply:
Yup. Any ideas how we could make this work, then? If SAML is configured but the expiry hasn't been set, it should default to 24h; if it wasn't configured and the expiry hasn't been set, it's 3m. If the expiry has been set, it should be whatever it's been set to.

Reply:
I think `PrepareSystemConfig` is the last thing that gets called to render the config in a way the service expects. You could put it in there